src/FITSIO.jl

```diff
@@ -166,31 +173,34 @@ mutable struct FITS
     memhandle::FITSMemoryHandle
     hidden::Any
 
-    function FITS(filename::AbstractString, mode::AbstractString="r")
-        f = (mode == "r"                      ? fits_open_file(filename, 0)    :
-             mode == "r+" && isfile(filename) ? fits_open_file(filename, 1)    :
-             mode == "r+"                     ? fits_create_file(filename)     :
-             mode == "w"                      ? fits_create_file("!"*filename) :
+    function FITS(filename::AbstractString, mode::AbstractString="r"; extendedparser = true)
+        openfn = extendedparser ? fits_open_file : fits_open_diskfile
+        createfn = extendedparser ? fits_create_file : fits_create_diskfile
+        f = (mode == "r"                      ? openfn(filename, 0)    :
+             mode == "r+" && isfile(filename) ? openfn(filename, 1)    :
+             mode == "r+"                     ? createfn(filename)     :
+             mode == "w"                      ?
+                (rm(filename, force = true); createfn(filename)) :
              error("invalid open mode: $mode"))
 
         new(f, filename, mode, Dict{Int, HDU}(), FITSMemoryHandle(), nothing)
     end
 
-    function FITS(data::Vector{UInt8}, mode::AbstractString="r", filename = "")
+    function FITS(data::Vector{UInt8}, mode::AbstractString="r", filename = ""; kwargs...)
```

Comment:
These keyword arguments are unused?

Reply:
Yes I wasn't sure what to do here. I have added `kwargs` here to be consistent with the signatures elsewhere. Now that I think about it perhaps it's better to remove this so that the user knows that `kwargs` are not being silently ignored. What do you think?

Reply:
Since the new keyword argument makes sense only when there is a filename argument, I don't think it's good to always accept any keyword arguments and silently discard them most of the time.  I'd rather not do this now, and if we later find that'd be a useful addition, we can do that in a non-breaking way (as opposed to removing a feature we deem to be not good, which would be a breaking change)

Reply:
I've removed this in the latest commit